src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Envelope.java

```diff
@@ -125,15 +131,22 @@ public String getZipFileName() {
         return zipFileName;
     }
 
-    public Event getLastEvent() {
-        return lastEvent;
+    public Event getStatus() {
+        return status;
     }
 
-    public void setLastEvent(Event lastEvent) {
-        this.lastEvent = lastEvent;
+    public void setStatus(Event lastEvent) {
+        this.status = lastEvent;
     }
 
     private void assignSelfToChildren(List<? extends EnvelopeAssignable> assignables) {
         assignables.forEach(assignable -> assignable.setEnvelope(this));
     }
+
+    @PrePersist
+    public void containerMustBePresent() {
+        if (container == null) {
+            throw new InvalidEnvelopeException("Missing required container");
```

Comment:
I think we should not do this because if the container is not set that is problem with the code and not the envelope itself. Can we just log instead ? If we find this in DB that could be reconciled anytime but if we do not process the document than that won't be correct.